src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -528,5 +528,45 @@ private static Vector<byte> GetVector(byte vectorByte)
                                                        0x01ul << 48) + 1;
         private const ulong byteBroadcastToUlong = ~0UL / Byte.MaxValue;
         private const ulong filterByteHighBitsInUlong = (byteBroadcastToUlong >> 1) | (byteBroadcastToUlong << (sizeof(ulong) * 8 - 1));
+        
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsSliceOf<T>(this Span<T> child, Span<T> parent) where T : struct
+        {
+            return child.Length <= parent.Length && IsSliceOf(
+                ref child.DangerousGetPinnableReference(), child.Length,
+                ref parent.DangerousGetPinnableReference(), parent.Length, out int ignored);
+        }
+
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        public static bool IsSliceOf<T>(this Span<T> child, Span<T> parent, out int start) where T : struct
+        {
+            start = -1;
+            return child.Length <= parent.Length && IsSliceOf(
+                ref child.DangerousGetPinnableReference(), child.Length,
+                ref parent.DangerousGetPinnableReference(), parent.Length, out start);
+        }
+
+        private static bool IsSliceOf<T>(ref T childRef, int childLength, ref T parentRef, int parentLength, out int start) where T : struct
+        {
+            long startBytesOffset = Unsafe.ByteOffset(ref parentRef, ref childRef).ToInt64();
+            long endBytesOffset = Unsafe.ByteOffset(
+                    ref Unsafe.Add(ref parentRef, parentLength), ref Unsafe.Add(ref childRef, childLength)
+                ).ToInt64();
```

Comment:
If these are moved separately then they are not part of the same array, so there shouldn't be any issue since they cannot overlap then. But this code could be made more efficient and with only one `ByteOffset` call, to simplify this and avoid issues like that.